python/ray/data/_internal/execution/interfaces/execution_options.py

```diff
@@ -50,6 +67,47 @@ def add(self, other: "ExecutionResources") -> "ExecutionResources":
             )
         return total
 
+    def subtract(self, other: "ExecutionResources") -> "ExecutionResources":
+        """Subtracts execution resources.
+
+        Returns:
+            A new ExecutionResource object with subtracted resources.
+        """
+        total = ExecutionResources()
+        if self.cpu is not None or other.cpu is not None:
+            total.cpu = (self.cpu or 0.0) - (other.cpu or 0.0)
+        if self.gpu is not None or other.gpu is not None:
+            total.gpu = (self.gpu or 0.0) - (other.gpu or 0.0)
+        if (
+            self.object_store_memory is not None
+            or other.object_store_memory is not None
+        ):
+            total.object_store_memory = (self.object_store_memory or 0.0) - (
+                other.object_store_memory or 0.0
+            )
+        return total
+
+    def max(self, other: "ExecutionResources") -> "ExecutionResources":
+        """Returns the maximum for each resource type."""
+        return ExecutionResources(
+            cpu=max(self.cpu or 0.0, other.cpu or 0.0),
+            gpu=max(self.gpu or 0.0, other.gpu or 0.0),
+            object_store_memory=max(
+                self.object_store_memory or 0, other.object_store_memory or 0
+            ),
+        )
+
+    def min(self, other: "ExecutionResources") -> "ExecutionResources":
+        """Returns the minimum for each resource type."""
+        return ExecutionResources(
+            cpu=min(self.cpu or float("inf"), other.cpu or float("inf")),
```

Comment:
Similar to `max()`, I'm wondering what the behavior should be if both `self` and `other` are `None`

```python
>>> ExecutionResources().min(ExecutionResources())
ExecutionResources(cpu=inf, gpu=inf, object_store_memory=inf)
```